server/src/main/java/org/apache/druid/indexing/overlord/IndexerMetadataStorageCoordinator.java

```diff
@@ -379,7 +381,8 @@ SegmentPublishResult commitAppendSegmentsAndMetadata(
       DataSourceMetadata startMetadata,
       DataSourceMetadata endMetadata,
       String taskAllocatorId,
-      @Nullable SegmentSchemaMapping segmentSchemaMapping
+      @Nullable SegmentSchemaMapping segmentSchemaMapping,
+      @Nullable String supervisorId
```

Comment:
Since the `startMetadata` and `endMetadata` are not nullable, `supervisorId` should not be nullable either. Caller should ensure passing a non-null value. DB layer can just do a validation that it is non null, it shouldn't need to determine the default value.